src/libstore/ca-specific-schema.sql

```diff
@@ -13,6 +13,19 @@ create table if not exists Realisations (
 
 create index if not exists IndexRealisations on Realisations(drvPath, outputName);
 
+-- We can end-up in a weird edge-case where a path depends on itself because
+-- it’s an output of a CA derivation, that happens to be the same as one of its
+-- dependencies.
+-- In that case we have a dependency loop (path -> realisation1 -> realisation2
+-- -> path) that we need to break by removing the dependencies between the
+-- realisations
+create trigger if not exists DeleteSelfRefsViaRealisations before delete on ValidPaths
+  begin
+    delete from RealisationsRefs where realisationReference = (
```

Comment:
I gave it a go on real system and it fails to delete entries still. I suggest the following change:

`delete from RealisationsRefs where realisationReference = ( ...` -> `delete from RealisationsRefs where realisationReference IN (...`. I think we expect multiple realizations to point to the same store path worst case. Not just one.

With `IN` I was able to collect all the garbage on my system \o/.